### ISSUE TITLE
508-defect-2: Tables are verbose and semantically mismatched on mobile

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#51164]
## Point of contact

joshkimux

### Blocked by Apple bug
- [AX: Tables with display properties are skipped, report wrong col/row counts](https://bugs.webkit.org/show_bug.cgi?id=257458)

## Severity level

2, Serious. Should be fixed in 1-2 sprints post-launch.

## Details

https://user-images.githubusercontent.com/14154792/209204376-84e27027-dc30-4f18-82ec-b80d914e9f9a.MP4

- On mobile using iOS / Safari / VoiceOver, content is too verbose. Columns and rows are provided after each string despite being designed as a list (visually at least). 
- An unknown string is announced after the date "Row 4, 9, 1, 2, 2, 2, 5, 2, 9... Column 1"

## Reproduction steps

1. Log in to staging view payments as user 228
2. Navigate to the responsive list view of the table
3. Swipe right and confirm column/row semantics are read out

## Proposed solution or next steps

If this is designed as a responsive list, it should read out as a list. For example, using a combination of headings and unordered lists.

It's unclear how to proceed with this ticket as issues may stem from the design system table.

- [ ] #55927
- [ ] If related to the design system, file a design system bug ticket
- [ ] If local to our team, remove table semantics and write as headings with unordered lists

## Type of issue

- [X] Screenreader
- [ ] Keyboard
- [ ] Focus
- [ ] Headings
- [ ] Color
- [ ] Zoom
- [X] Semantics
- [ ] Axe-core
- [ ] Something else

**Comment by jaredcunha** [comment]
Something is really strange. I don't understand why these padding highlights have a weird diagonal. This is from plugging my phone into my computer.

![IMG_3287](https://github.com/department-of-veterans-affairs/va.gov-team/assets/1094951/95839e6a-8fe3-4a4c-b695-1f43c4069f89)

It seems to be using the responsive table
<img width="880" alt="Screenshot 2023-05-09 at 12 26 50 PM" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/1094951/51383abc-3a0d-45e8-a8b5-4df285a056ac">

seems to be using this component: https://github.com/department-of-veterans-affairs/component-library/blob/main/packages/react-components/src/components/Table/Table.jsx

Wonder if it's worth seeing if there are similar issues with other instances of the table
<img width="910" alt="Screenshot 2023-05-09 at 12 33 58 PM" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/1094951/5861b791-1ed5-4d81-85a9-7edb0bebfe92">
